dlt/sources/helpers/rest_client/auth.py

```diff
@@ -104,16 +88,8 @@ def __call__(self, request: PreparedRequest) -> PreparedRequest:
 
 @configspec
 class HttpBasicAuth(AuthConfigBase):
-    type: Final[Literal["http"]] = "http"  # noqa: A003
-    scheme: Literal["basic"] = "basic"
-    username: str
-    password: TSecretStrValue
-
-    def __init__(
-        self, username: str = config.value, password: TSecretStrValue = secrets.value
-    ) -> None:
-        self.username = username
-        self.password = password
+    username: str = ""
```

Comment:
it is better to set `username` to None. this makes it a required config field. "" makes it optional (None is used to detect that field is missing)

Reply:
Fixed in b9bb0d354aa281fb84838f39d7e3f25e7fe9b040